From: Debian Samba Maintainers <pkg-samba-maint@lists.alioth.debian.org>
Date: Mon, 29 Feb 2016 12:04:32 -0500
Subject: bug_601406_fix-perl-path-in-example

---
 examples/misc/wall.perl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/misc/wall.perl b/examples/misc/wall.perl
index 9303658..72d6e89 100644
--- a/examples/misc/wall.perl
+++ b/examples/misc/wall.perl
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 # 
 #@(#) smb-wall.pl Description:
 #@(#) A perl script which allows you to announce whatever you choose to
