From: Debian Samba Maintainers <pkg-samba-maint@lists.alioth.debian.org>
Date: Mon, 29 Feb 2016 12:04:33 -0500
Subject: waf_smbpasswd_location

---
 dynconfig/wscript | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dynconfig/wscript b/dynconfig/wscript
index dba74f6..2d17e7c 100755
--- a/dynconfig/wscript
+++ b/dynconfig/wscript
@@ -254,6 +254,8 @@ dynconfig = {
     'SMB_PASSWD_FILE' : {
          'STD-PATH':  '${PRIVATE_DIR}/smbpasswd',
          'FHS-PATH':  '${PRIVATE_DIR}/smbpasswd',
+         'OPTION':    '--with-smbpasswd-file',
+         'HELPTEXT':  'Where to put the smbpasswd file',
          'DELAY':     True,
     },
 }
