From: Ashok Ramakrishnan <ashok.ramakrishnan@simplivity.com>
Date: Tue, 16 Aug 2016 13:36:50 +0000
Subject: HYP-none : Changes to support debug build.

---
 source3/smbd/smb2_create.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source3/smbd/smb2_create.c b/source3/smbd/smb2_create.c
index 75986c8..92f1ebc 100644
--- a/source3/smbd/smb2_create.c
+++ b/source3/smbd/smb2_create.c
@@ -388,7 +388,7 @@ static NTSTATUS smbd_smb2_create_durable_lease_check(
 		return NT_STATUS_OBJECT_NAME_NOT_FOUND;
 	}
 
-	unix_syntax_filename = strdup(requested_filename);
+	unix_syntax_filename = SMB_STRDUP(requested_filename);
 
 	if (unix_syntax_filename == NULL) {
 		DEBUG(1, ("smbd_smb2_create_durable_lease_check: can't dup filename; out of memory\n"));
