From: Ashok Ramakrishnan <ashok.ramakrishnan@simplivity.com>
Date: Wed, 17 Aug 2016 13:23:38 +0000
Subject: HYP-none : Disable mmap by default in tdbtool

---
 lib/tdb/tools/tdbtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tdb/tools/tdbtool.c b/lib/tdb/tools/tdbtool.c
index beb3af1..0bfd280 100644
--- a/lib/tdb/tools/tdbtool.c
+++ b/lib/tdb/tools/tdbtool.c
@@ -35,7 +35,7 @@ int bIterate = 0;
 char *line;
 TDB_DATA iterate_kbuf;
 char cmdline[1024];
-static int disable_mmap;
+static int disable_mmap = 1;
 static int disable_lock;
 
 enum commands {
