From: Paul Cerqua <paul.cerqua@simplivity.com>
Date: Fri, 2 Sep 2016 18:02:26 +0000
Subject: HYP-1934:  Getting smbtorture lease tests to pass again

---
 source3/smbd/smb2_create.c | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/source3/smbd/smb2_create.c b/source3/smbd/smb2_create.c
index 92f1ebc..f41e663 100644
--- a/source3/smbd/smb2_create.c
+++ b/source3/smbd/smb2_create.c
@@ -961,20 +961,21 @@ static struct tevent_req *smbd_smb2_create_send(TALLOC_CTX *mem_ctx,
 				 * 4) then, durable and lease context requests should be ignored.
 				 */
 				durable_requested = false;
+#ifdef MICROSOFT_FRAMEWORK_TESTS
 				requested_oplock_level = SMB2_OPLOCK_LEVEL_NONE;
 				lease_ptr = NULL;
-			} else {
+#endif
+			}
 
-				if (!smb2_lease_key_valid(&lease.lease_key)) {
-					lease_ptr = NULL;
-					requested_oplock_level = SMB2_OPLOCK_LEVEL_NONE;
-				}
+			if (!smb2_lease_key_valid(&lease.lease_key)) {
+				lease_ptr = NULL;
+				requested_oplock_level = SMB2_OPLOCK_LEVEL_NONE;
+			}
 
-				if ((smb2req->xconn->protocol < PROTOCOL_SMB3_00) &&
-						(lease.lease_version != 1)) {
-					DEBUG(10, ("v2 lease key only for SMB3\n"));
-					lease_ptr = NULL;
-				}
+			if ((smb2req->xconn->protocol < PROTOCOL_SMB3_00) &&
+					(lease.lease_version != 1)) {
+				DEBUG(10, ("v2 lease key only for SMB3\n"));
+				lease_ptr = NULL;
 			}
 		}
 
