From: Ashok Ramakrishnan <ashok.ramakrishnan@simplivity.com>
Date: Fri, 21 Jul 2017 14:47:41 -0400
Subject: Use a different file name for temp file.

---
 source3/smbd/smbXsrv_open.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source3/smbd/smbXsrv_open.c b/source3/smbd/smbXsrv_open.c
index 043d27a..9827d1e 100644
--- a/source3/smbd/smbXsrv_open.c
+++ b/source3/smbd/smbXsrv_open.c
@@ -342,7 +342,7 @@ fail:
 NTSTATUS smbXsrv_open_global_init(void)
 {
         char *global_path = NULL;
-        const char *tmp_path="/etc/samba/smbXsrv_open_global.tdb";
+        const char *tmp_path="/etc/samba/smbxsrv_open_global.tdb";
         struct db_context *db_ctx = NULL;
         NTSTATUS status = NT_STATUS_OK;
         int index,saved_index;
